### ISSUE TITLE
Error loading pandoc.vim with Vim 8 package loader

### ISSUE THREAD

**Issue by benwebber** [vim-pandoc/vim-pandoc#364]
I recently switched to Vim 8's built-in package loader (`:help packages`). As of 7fe4ae88cb955d46fa4bc8adc23a9ddb2c14bec5 (#352), I receive the following message on startup:

```
Error detected while processing /path/to/vim/dir/pack/plugins/start/vim-pandoc/plugin/pandoc.vim:
line   79:
E121: Undefined variable: g:pandoc#filetypes#pandoc_markdown
E15: Invalid expression: g:pandoc#filetypes#pandoc_markdown == 1
Press ENTER or type command to continue
```

It seems like Vim loads `ftdetect` scripts *after* scripts in the `plugin` directory, so the variable hasn't been defined yet.

Here are the relevant lines from `:scriptnames`:

```
 60: ~/.local/share/vim/pack/plugins/start/vim-pandoc/plugin/pandoc.vim
 61: ~/.local/share/vim/pack/plugins/start/vim-pandoc/ftdetect/pandoc.vim
 62: ~/.local/share/vim/pack/plugins/start/vim-pandoc-syntax/plugin/pandoc-syntax-check.vim
```

The Vim documentation (`:help initialization`) doesn't mention when Vim loads `ftdetect` scripts, only that it tries to source `plugin` scripts found in the `runtimepath` or `packpath`. I haven't been able to find any information on how to change this order.

If I revert to 2bbb80576a62f58417e05d8600bb5647868bedf7, the plugin loads without complaints. For now I've defined `g:pandoc#filetypes#pandoc_markdown` in my `$VIMRC`. I can also work around the issue indirectly by explicitly loading `ftdetect` scripts at the end of my `$VIMRC`.

I know next to nothing about Vim plugin internals. Is there a way to structure the plugin so that it's compatible with the default Vim 8 package loader?

**Comment by alerque** [comment]
Thanks for this report, and I'm sorry you're having trouble. This is a known problem we've been bouncing around for a few days. I fixed one issue the other day and caused this one for others. What plugin manager you use is part of the equation, but _shouldn't_ matter. We'll post a fix for this soon, I just keep getting distracted (world wide problem).

See also #363.

**Comment by alerque** [comment]
Fixed by merge of #363.
